src/test/java/org/kiwiproject/dropwizard/util/health/keystore/ExpiringKeystoreHealthCheckTest.java

```diff
@@ -0,0 +1,267 @@
+import static org.assertj.core.api.Assertions.entry;
+import static org.kiwiproject.collect.KiwiLists.first;
+
+import io.dropwizard.util.Duration;
+import org.assertj.core.api.SoftAssertions;
+import org.assertj.core.api.junit.jupiter.SoftAssertionsExtension;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.api.io.TempDir;
+import org.kiwiproject.base.process.ProcessHelper;
+import org.kiwiproject.dropwizard.util.config.KeystoreConfig;
+import org.kiwiproject.io.KiwiIO;
+import org.kiwiproject.security.KeyStoreType;
+
+import java.nio.file.Path;
+import java.security.KeyStore;
+import java.security.KeyStoreException;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+@DisplayName("ExpiringKeystoreHealthCheck")
+@ExtendWith(SoftAssertionsExtension.class)
+class ExpiringKeystoreHealthCheckTest {
+
+    private static final DateTimeFormatter DATE_FORMATTER = DateTimeFormatter.ofPattern("EE, d LLL yyyy");
+
+    @Nested
+    class ShouldReportUnhealthy {
+        @SuppressWarnings("unchecked")
+        @Test
+        void whenThereAreExpiredCerts(SoftAssertions softly, @TempDir Path tempDir) throws Exception {
+            var dn = "CN=Unit Test, OU=Development, O=Project, L=Here, ST=VA, C=US";
+            var path = createTemporaryKeystore(tempDir, 30, dn);
+
+            var config = KeystoreConfig.builder()
+                    .name("Test Keystore Config")
+                    .path(path)
+                    .pass("unittest")
+                    .ttl(Duration.days(45))
+                    .build();
+
+            var healthCheck = new ExpiringKeystoreHealthCheck(config);
+            var result = healthCheck.check();
+
+            softly.assertThat(result.isHealthy()).isFalse();
+            softly.assertThat(result.getMessage()).isEqualTo(path + ": 1 expired certs, 0 certs expiring soon");
+            softly.assertThat(result.getError()).isNull();
+
+            var details = result.getDetails();
+            softly.assertThat(details).contains(
+                    entry("path", path),
+                    entry("ttl", Duration.days(45))
+            );
+
+            List<BasicCertInfo> validCerts = getBasicCertInfoList(details, "validCerts");
+            softly.assertThat(validCerts).isEmpty();
+
+            List<BasicCertInfo> expiredCerts = getBasicCertInfoList(details, "expiredCerts");
+            softly.assertThat(expiredCerts).hasSize(1);
+
+            var certInfo = first(expiredCerts);
+            softly.assertThat(certInfo.getSubjectDN()).isEqualTo("CN=Unit Test, OU=Development, O=Project, L=Here, ST=VA, C=US");
+            softly.assertThat(certInfo.getIssuerDN()).isEqualTo("CN=Unit Test, OU=Development, O=Project, L=Here, ST=VA, C=US");
```

Comment:
Unless I keep misreading, the subject and issuer DN are the same...was that intented?

Reply:
For tests it is a self signed cert